### ISSUE TITLE
Use same GitHub token logic as gh

### ISSUE THREAD

**Issue by maelle** [ropensci-review-tools/pkgcheck#153]
or even the gh function?

Asking this as I am getting a 401.

**Comment by maelle** [comment]
after getting a 401 for I think an expired GITHUB_GRAPHQL_TOKEN I had in my .Renviron I ran `Sys.setenv ("GITHUB_TOKEN" =gh::gh_token())` then got an error about having several tokens. I tried setting GITHUB_GRAPHQL_TOKEN to "" but it did not work so I commented it out in my .Renviron then restarted R then ran `Sys.setenv ("GITHUB_TOKEN" =gh::gh_token())` then it worked.

**Comment by mpadge** [comment]
Yes indeed, that should be done. The reason not is that i don't use the git credentials store, because i have my own wacky encrypted authorization system so i can push to multiple remotes with one command. But my idiosyncracies shouldn't make it more difficult for other people to use this package. I'll update my own workflow, and then get to this straight after that. Thanks!

**Comment by maelle** [comment]
In that case you'd just need to replace the current function with `gh::gh_token()`.

Now I am curious about your complex workflow, sounds like a good tech note topic. :grin:

**Comment by mpadge** [comment]
> Now I am curious about your complex workflow, sounds like a good tech note topic. grin

No, it's not, and not recommended for anybody else. It was just a fun coding exercise. Also because i like having a pasword prompt after `git push`. That microsecond delay and mental prompt has saved me from many silly pushes. So my `git push` works like old-school password-style, but is all 2FA via ssh which shouldn't need a password. It's silly, really, but i like it.

**Comment by PietrH** [comment]
What about using gitcreds like usethis? 

https://gitcreds.r-lib.org/

I don't like the idea of someone putting their github token in plain text in .Renviron or Sys env. 

Usethis actually warns explicitly not to do this: https://usethis.r-lib.org/articles/git-credentials.html?q=gitcreds#put-your-pat-into-the-local-git-credential-store

While pkgcheck asks users to do it on the frontpage: https://docs.ropensci.org/pkgcheck/#setup

**Comment by mpadge** [comment]
Thanks @PietrH, sounds fair enough. The only trick here is that everything has to still work with environment variables, because that's the only stable way to build all this inside a Docker container. But previous issues with inter-compatibility of the two approaches are resolved now, so we can easily implement and recommend use of gitcreds here while (mostly invisibly) retaining use of `Sys.env`.

**Comment by PietrH** [comment]
What needs to happen to implement this? Maybe I can help out.

**Comment by mpadge** [comment]
Thanks @PietrH! I was thinking along exactly those lines. My suggestion was going to be to organise a co-working session where anybody who is interested can learn how to develop new checks here. The more checks we implement, the less I am able to maintain everything here, so I am particularly keen to onboard new co-maintainers. Anybody who has implemented one check will quickly understand most of the package. What do you think? @maelle: your thoughts?

**Comment by maelle** [comment]
But this issue is not about a check? It's about finding where in the codebase tokens are handled+documented, and replacing that with gitcreds?

Otherwise I agree with the idea of a coworking session, but do you mean an actual coworking session (@steffilazerte) or "just" a zoom call?

**Comment by mpadge** [comment]
> But this issue is not about a check? It's about finding where in the codebase tokens are handled+documented, and replacing that with gitcreds?

I view it as a check, in that `pkgcheck` would just check whether envvars are used directly, and link to sensible explanation of why and how to replace with gitcreds (or whatever else). It would check "handled" but not necessarily "documented", and I think that's okay here.

> Otherwise I agree with the idea of a coworking session, but do you mean an actual coworking session ([@steffilazerte](https://github.com/steffilazerte)) or "just" a zoom call?

Dunno. Whatever would work best. I guess the latter, as some kind of specific session to guide people through the process, and then have small groups each tackle one of the outstanding check ideas.

**Comment by maelle** [comment]
What are the next action points for this issue? Initially I was only suggesting using the same logic/code as gh for finding the token to make a GH API request. If you use  gh under the hood anyway, no need for anything on top of it here, since gh too supports override via an environment variable?

**Comment by mpadge** [comment]
`gh` token handling is now implicit throughout
